pyflakes/test/test_imports.py

```diff
@@ -564,6 +563,15 @@ def bar():
         fu
         ''', m.UnusedImport, m.UndefinedName)
 
+    def test_shadowedInNestedScope(self):
+        self.flakes('''
+        import fu
+        def bar():
+            import fu
+            fu
+        fu
+        ''', m.ImportShadowedByImport)
```

Comment:
I'm not sure I agree this is a desirable behavior. I know there's also an `ImportShadowedByLoopVar` ... I'm not sure I agree with that either. Citing the [design principles](https://github.com/pyflakes/pyflakes#design-principles):

> Pyflakes makes a simple promise: it will never complain about style, and it will try very, very hard to *never emit false positives*.

There's nothing inherently erroneous about the code here. Can we be sure that no one will ever have a legitimate reason to do this? What if we change the example a bit:

```python
from StringIO import StringIO

def bar():
    from cStringIO import StringIO
    StringIO

StringIO
```